akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ClusterShardingImpl.scala

```diff
@@ -305,8 +302,10 @@ import akka.util.Timeout
     with scaladsl.EntityRef[M]
     with InternalRecipientRef[M] {
 
-  override def tell(msg: M): Unit =
+  override def tell(msg: M): Unit = {
+    if (msg == null) throw InvalidMessageException("[null] is not an allowed message")
     shardRegion ! ShardingEnvelope(entityId, msg)
```

Comment:
An alternative could be to put that null check in the constructor of `ShardingEnvelope`. Then it would catch some more cases.

Reply:
That's could be done.